test_info_repr failing locally

OSX py2.7

```
def test_info_repr(self):
        max_rows = 60
        max_cols = 20
        # Long
        h, w = max_rows + 1, max_cols - 1
        df = DataFrame({k: np.arange(1, 1 + h) for k in np.arange(w)})
        assert has_vertically_truncated_repr(df)
        with option_context('display.large_repr', 'info'):
            assert has_info_repr(df)
    
        # Wide
        h, w = max_rows - 1, max_cols + 1
        df = DataFrame({k: np.arange(1, 1 + h) for k in np.arange(w)})
>       assert has_horizontally_truncated_repr(df)
E       assert False
E        +  where False = has_horizontally_truncated_repr(    0   1   2   3   4   5   6   7   8   9   10  11  12  13  14  15  16  17  18... 59  59  59  59  59  59  59  59  59  59  59  59  59  59  59  59  59  59  59  59)

pandas/tests/io/formats/test_format.py:1601: AssertionError
```

--- a/pandas/tests/io/formats/test_format.py
+++ b/pandas/tests/io/formats/test_format.py
@@ -1586,8 +1586,12 @@ c  10  11  12  13  14\
             assert '...' in df._repr_html_()
 
     def test_info_repr(self):
+        # GH#21746 For tests inside a terminal (i.e. not CI) we need to detect
+        # the terminal size to ensure that we try to print something "too big"
+        term_width, term_height = get_terminal_size()
+
         max_rows = 60
-        max_cols = 20
+        max_cols = 20 + (max(term_width, 80) - 80) // 4
         # Long
         h, w = max_rows + 1, max_cols - 1
         df = DataFrame({k: np.arange(1, 1 + h) for k in np.arange(w)})

